php/utils-wp.php

```diff
@@ -319,11 +319,29 @@ function wp_get_table_names( $args, $assoc_args = array() ) {
 	global $wpdb;
 
 	$tables = array();
+
+	// Pre-load tables SQL query with Views restriction if needed.
+	if ( get_flag_value( $assoc_args, 'base-tables-only' ) ) {
+		$tables_sql = $wpdb->prepare( 'SHOW FULL TABLES WHERE Table_Type = %s', 'BASE TABLE' );
+
+	} elseif ( get_flag_value( $assoc_args, 'views-only' ) ) {
+		$tables_sql = $wpdb->prepare( 'SHOW FULL TABLES WHERE Table_Type = %s', 'VIEW' );
+
+	}
+
 	if ( get_flag_value( $assoc_args, 'all-tables' ) ) {
-		$tables = $wpdb->get_col( 'SHOW TABLES' );
+		if ( empty( $tables_sql ) ) {
+			$tables_sql = 'SHOW TABLES';
+		}
+		$tables = $wpdb->get_col( $tables_sql, 0 ); // WPCS: unprepared SQL OK.
 
 	} elseif ( get_flag_value( $assoc_args, 'all-tables-with-prefix' ) ) {
-		$tables = $wpdb->get_col( $wpdb->prepare( 'SHOW TABLES LIKE %s', esc_like( $wpdb->get_blog_prefix() ) . '%' ) );
+		if ( empty( $tables_sql ) ) {
+			$tables_sql = $wpdb->prepare( 'SHOW TABLES LIKE %s', esc_like( $wpdb->get_blog_prefix() ) . '%' );
+		} else {
+			$tables_sql .= $wpdb->prepare( ' AND Tables_in_$wpdb->dbname LIKE %s', esc_like( $wpdb->get_blog_prefix() ) . '%' );
```

Comment:
The SQL string seems broken here. This should be constructed similarly to line 353/371 below:

https://github.com/wp-cli/wp-cli/pull/5128/files#diff-db14fab272c9fba408859cf06e743d6fR371

Reply:
There may well have been a bug in what I wrote, although the principle works. However perhaps it's not the best way to do it, and changing as suggested certainly gives internal consistency in the function.